CONTRIBUTING.md

```diff
@@ -62,3 +110,23 @@ Thanks for taking the plunge!
   rather than `[1., 2.]`
 * In docstrings, optional arguments, including separators and spaces, are surrounded by brackets,
   e.g. `mymethod(required[, optional1[, optional2] ]; kwargs...)`
+
+## Git Recommendations For Pull Requests
+
+* Avoid working from the `main` branch of your fork, creating a new branch will make it easier if DataFrame's `main` branch changes and you need to update your pull request.
+
+* Try to [squash](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) together small commits that make repeated changes to the same section of code so your pull request is easier to review, and DataFrame's won't have any broken intermediate commits. A reasonable number of separate well-factored commits is fine, especially for larger changes.
+
+* If any conflicts arise due to changes in DataFrame's `main` branch, prefer updating your pull request branch with `git rebase` versus `git merge` or `git pull`, since the latter will introduce merge commits that clutter the git history with noise that makes your changes more difficult to review.
```

Comment:
```suggestion
* If any conflicts arise due to changes in DataFrame.jl `main` branch, prefer updating your pull request branch with `git rebase` (rather than `git merge`), since the latter will introduce a merge commit that might confuse GitHub when displaying the diff of your PR, which makes your changes more difficult to review. Alternatively use conflict resolution tool available at GitHub.
```